CMake/AbseilHelpers.cmake

```diff
@@ -107,6 +113,7 @@ function(absl_cc_library)
       add_library(${_NAME} STATIC "")
       target_sources(${_NAME} PRIVATE ${ABSL_CC_LIB_SRCS} ${ABSL_CC_LIB_HDRS})
       target_include_directories(${_NAME}
+        ${ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS_KEYWORD}
         PUBLIC
           $<BUILD_INTERFACE:${ABSL_COMMON_INCLUDE_DIRS}>
           $<INSTALL_INTERFACE:${ABSL_INSTALL_INCLUDEDIR}>
```

Comment:
This should be:
```cmake
      target_include_directories(${_NAME}
        PUBLIC
          $<BUILD_INTERFACE:${ABSL_COMMON_INCLUDE_DIRS}>
      )

      target_include_directories(${_NAME} ${ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS_KEYWORD}
        PUBLIC
          $<INSTALL_INTERFACE:${ABSL_INSTALL_INCLUDEDIR}>
      )
```
The second `target_include_directories` is actually equivalent to `target_...(... SYSTEM INTERFACE ...)`
The same applies to the other `target_include_directories` lower.

This allows having warnings inside the Abseil project, but remove them for the users of the interface(for the actual users that can't modify the code and should not see the warnings).
With this change, the option can be removed and `SYSTEM` used directly.

Reply:
@lilianmoraru if we are building Abseil as a subproject within a larger project (using e.g. CMake's `add_subdirectory`), we want to be able to silence these warnings as well. This is why `SYSTEM` should go into build interface as well if the user so desires.